### PR TITLE
WT-2233 Base aggressive eviction on the number of pages being evicted.

### PR DESCRIPTION
**Pull request by agorrod**
Not the number of loops since we hit the eviction target.

### CODE REVIEW

**Comment by agorrod** [comment]
@keithbostic @michaelcahill This pull request is intended as a discussion point. It is too invasive to be merged as is without more testing (aggressive eviction is enabled in a lot of different places).

Having said that, it improves the behavior of the LAS file in the ``repro_update.js`` reproducer from SERVER-21585 for me. Without this change the LAS file grows to ~300MB after less than 10 minutes using the above reproducer. With this change the file grows to about 4.5MB.

**Comment by agorrod** [comment]
This needs a workload that demonstrates it's benefit.

**Comment by agorrod** [comment]
I switched the Jenkins performance jobs to test this branch. The performance differences are:

 Configuration | Old | New | change
------------ | ------------- | ----- | --------
Medium btree read | 40 million | 60 million | +50%
Update btree read | 40 million | 47 million | +18%
Medium multi LSM read | 37 million | 34 million | -8%
Evict btree read | 50 million | 48 million  | -4%
Evict LSM read | 50 million | 108 million | +116%

I'd say this change warrants further investigation.

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -473,6 +474,21 @@ __evict_update_work(WT_SESSION_IMPL *session)
...
+	 * Update the eviction progress tracking.
+	 * TODO: Don't use statistics for this tracking, but it's nice to have
+	 * the statistics performant counters.
+	 */
+	evict_count = WT_STAT_READ(conn->stats, cache_eviction_clean) +
+	    WT_STAT_READ(conn->stats, cache_eviction_dirty);
+	if (evict_count > cache->saved_evict_count)
+		evict_count = evict_count - cache->saved_evict_count;
+	else
+		evict_count = 0;
```
Comment (sueloverso) [inline src/evict/evict_lru.c]: Please add a comment about what the progress is computing.  Between the use of previous values, resetting evict_count and hard-coded 90%, it is hard to know exactly what sort of number you're looking for.

**Comment by agorrod** [comment]
This isn't ready for prime time. Needs more work and verification before going down this path. Closing the pull request for now, but leaving the branch open.

**Pull request closed without merging**